src/it/scala/net/snowflake/spark/snowflake/PushdownEnhancement02.scala

```diff
@@ -455,6 +457,58 @@ class PushdownEnhancement02 extends IntegrationSuiteBase {
     }
   }
 
+  test("test pushdown number functions: PI() and Round()/Random") {
+    // Don't run test with use_copy_unload because COPY UNLOAD converts
+    // PI value 3.141592653589793 to 3.141592654
+    if (!params.useCopyUnload) {
+      jdbcUpdate(s"create or replace table $test_table_number " +
+        s"(d1 decimal(38, 10), f1 float)")
+      jdbcUpdate(s"insert into $test_table_number values " +
+        s"(-1.9, -1.9),  (-1.1, -1.1), (0, 0), (1.1, 1.1), (1.9, 1.9)")
+
+      val tmpDF = sparkSession.read
+        .format(SNOWFLAKE_SOURCE_NAME)
+        .options(thisConnectorOptionsNoTable)
+        .option("dbtable", test_table_number)
+        .load()
+
+      tmpDF.createOrReplaceTempView("test_table_number")
+
+      val resultDF =
+        sparkSession
+          .sql(s"select round(d1), round(f1), PI()" +
+            " from test_table_number")
+
+      resultDF.printSchema()
+      resultDF.show(10, false)
+
+      val PI = 3.141592653589793
+      val expectedResult = Seq(
+        Row(BigDecimal(-2), (-2).toDouble, PI),
+        Row(BigDecimal(-1), (-1).toDouble, PI),
+        Row(BigDecimal(0), (0).toDouble, PI),
+        Row(BigDecimal(1), (1).toDouble, PI),
+        Row(BigDecimal(2), (2).toDouble, PI)
+      )
+
+      testPushdown(
+        s"""SELECT ( ROUND ( "SUBQUERY_0"."D1" , 0 ) ) AS "SUBQUERY_1_COL_0",
+           |( ROUND ( "SUBQUERY_0"."F1" , 0 ) ) AS "SUBQUERY_1_COL_1",
+           |( 3.141592653589793 ) AS "SUBQUERY_1_COL_2" FROM
+           |( SELECT * FROM ( $test_table_number ) AS
+           |"SF_CONNECTOR_QUERY_ALIAS" ) AS "SUBQUERY_0"
+           |""".stripMargin,
+        resultDF,
+        expectedResult
+      )
```

Comment:
Maybe we can change this testPushdown function to take regex matches as ground truth, so that if there is uncertainties in the executed query like random(), we can use regex to match it. Just my random thoughts, maybe there is not much use cases for the effort.

Reply:
It's good idea. But so far, it is no necessary. Let's do it, when necessary.